File: src/ObjectManager/Object.Tes/FilePacks/Records/005-AACT.Action.cs
using OA.Core;
using System;

namespace OA.Tes.FilePacks.Records
{
    public class AACTRecord : Record
    {
        public override string ToString() => $"AACT: {EDID.Value}";
        public STRVField EDID; // Action Name
        public CREFField CNAME; // RGB color

        public override bool CreateField(UnityBinaryReader r, GameFormatId formatId, string type, uint dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = new STRVField(r, dataSize); return true;
                case "CNAME": CNAME = new CREFField(r, dataSize); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/005-AACT.Action.swift
//
//  AACTRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class AACTRecord: Record {
    public var description: String { return "AACT: \(EDID)" }
    public var EDID: STRVField // Editor ID
    public var CNAME: CREFField // RGB color

    public override bool CreateField(UnityBinaryReader r, GameFormatId formatId, string type, uint dataSize)
    {
        switch (type)
        {
            case "EDID": EDID = new STRVField(r, dataSize); return true;
            case "CNAME": CNAME = new CREFField(r, dataSize); return true;
            default: return false;
        }
    }
}
